tests/lib/c_lib/src/main.c

```diff
@@ -558,6 +558,45 @@ void test_tolower_toupper(void)
 	zassert_equal(strcmp(lw, tolw), 0, "tolower error");
 }
 
+void test_strsep_do(char *str, char *sep, int tlen, char **toks, bool expect)
+{
+	int len = 0;
+	char *p, *tok, buf[64+1] = {0};
+
+	strncpy(buf, str, 64);
+	p = buf;
+
+	while (len < tlen && (tok = strsep(&p, sep)) != NULL) {
+		printk("tok: '%s' sep: '%s'\n", tok, toks[len]);
+		if (strcmp(tok, toks[len]) != 0) {
+			break;
+		}
+		len++;
+	}
+	if (expect) {
+		zassert_equal(len, tlen,
+			      "strsep error '%s' / '%s'", str, sep);
+	} else {
+		zassert_not_equal(len, tlen,
+				  "strsep error '%s' / '%s'", str, sep);
+	}
+}
+
+void test_strsep(void)
+{
+	const char **tc01 = { "1", "2", "3", "4", "5" };
```

Comment:
You can't initialize a pointer this way.  It needs to be:

    const char *tc01[] = { "1", "2", "3", "4", "5" };

Reply:
Sorry. That was a sleepy check checkpatch fixup. Will correct it.